src/trix/elements/trix_editor_element.js

```diff
@@ -161,6 +161,19 @@ installDefaultCSSForTagName("trix-editor", `\
 }`)
 
 export default class TrixEditorElement extends HTMLElement {
+  static get formAssociated() {
+    return config.editor.formAssociated
+  }
```

Comment:
@jorgemanrubia I've re-purposed this PR to re-introduce `ElementInternals` alongside Constraint validation integration in a backwards compatible way.

Automatic support inference hinges on `"ElementInternals" in window`. That assignment occurs in a new `Trix.config.editor.formAssociated` value. 

Is introducing a new configuration better than documenting that application code can assign `Trix.elements.TrixEditorElement.formAssociated` directly?

Does the scope of this diff feel appropriate, or would it be better to split out a separate PR that introduces the configuration (or documents the `Trix.elements.TrixEditorElement.formAssociated` property, if that feels better) to ship ahead of the constraint validation and `[disabled]` support?

Reply:
Thanks for working on this @seanpdoyle. The scope and implementation are great. My only doubt is about exposing this as something you want to disable. In which scenarios would a Trix user want to disable this? I love that it will just disable it internally when not supported, but I wonder why would a customer want to disable that. I prefer not to add more user-facing config options unless justified.

Reply:
> In which scenarios would a Trix user want to disable this?

I was considering the scenario you've outlined where an application's backwards compatibility commitments mean that they support browsers that do not support `ElementInternals`. 

If a team were supporting Safari 14, but rendered `trix-editor[required]` or `trix-editor[disabled]`, it'd behave as expected for users issuing requests from modern browsers, while those attributes would be ignored for the ones with legacy browsers.

If an application disabled the enhancements entirely, then their application could be forced to target the older browsers for support (by disabling any new functionality like Constraint validation).

> I prefer not to add more user-facing config options unless justified.

If conditional support without user-facing doesn't feel risky, I'm happy to revert the configuration and CI seams.

Reply:
I've reverted the `Trix.config.editor` changes, and changed the guidance in the `README.md` to encourage `Trix.elements.TrixEditorElement.formAssociated = false` as the supported escape hatch.